src/bun.js/bindings/webcore/JSDOMException.cpp

```diff
@@ -129,9 +129,32 @@ template<> JSC::EncodedJSValue JSC_HOST_CALL_ATTRIBUTES JSDOMExceptionDOMConstru
     EnsureStillAliveScope argument0 = callFrame->argument(0);
     auto message = argument0.value().isUndefined() ? emptyString() : convert<IDLDOMString>(*lexicalGlobalObject, argument0.value());
     RETURN_IF_EXCEPTION(throwScope, {});
+
+    String name = "Error"_s;
+    JSValue cause = {};
+
     EnsureStillAliveScope argument1 = callFrame->argument(1);
-    auto name = argument1.value().isUndefined() ? "Error"_s : convert<IDLDOMString>(*lexicalGlobalObject, argument1.value());
-    RETURN_IF_EXCEPTION(throwScope, {});
+    if (argument1.value().isObject()) {
+        JSObject* optionsObj = asObject(argument1.value());
+
+        // Get name from options
+        JSValue nameValue = optionsObj->get(lexicalGlobalObject, vm.propertyNames->name);
+        RETURN_IF_EXCEPTION(throwScope, {});
+        if (!nameValue.isUndefined())
+            name = convert<IDLDOMString>(*lexicalGlobalObject, nameValue);
+        RETURN_IF_EXCEPTION(throwScope, {});
+
+        // Get cause from options
+        if (optionsObj->hasProperty(lexicalGlobalObject, vm.propertyNames->cause)) {
```

Comment:
`getIfPropertyExists` calls `has` and `get` in one go

```suggestion
        auto causeValue = optionsObj->getIfPropertyExists(lexicalGlobalObject, vm.propertyNames->cause);
        RETURN_IF_EXCEPTION(throwScope, {});
        if (causeValue) {
```